### DISCUSSION TITLE
Authelia for Django oath2?

### DISCUSSION THREAD

**Discussion by tetricky** [authelia/authelia#4884] [Q&A]
I appreciate this is not directly project related, but I hope it's appropriate to ask here. I'm trying to use authelia for authentication and user login to [misago](https://misago-project.org), a django (I think) based forum application. I don't really understand the nuts and bolts, but have a working system with other integrations (hedgedoc, gitea). I get a sense that authelia can provide what is needed here, and I can get the authorisation link and the redirect, but I can't seem to get the magic juju to get misago to read the token. I have authelia with an lldap backend, and caddy2 as the reverse proxy.

The guidance for connecting to misago to an oauth2 provider is [here](https://misago-project.org/t/oauth-2-client-configuration-guide/1147/).

Any advice or pointers much appreciated.

**Comment by james-d-elliott** [comment]
Might want to include logs from the app so we can see what's going on

**Comment by tetricky** [comment]
There are no useful logs, unless I reconfigure with debug. Which is problematic. What I need to understand is how to get the access token. I'm using the url https://my.authelia.domain/api/oidc/token. Does it require get or post? How should the grant be sent to it? What does the JSON response from Authelia look like?

My issue is how to configure the 'Access token configuration', and the 'User data retrieval from the OAuth 2 server' from the [configuration above](https://misago-project.org/t/oauth-2-client-configuration-guide/1147/).

**Reply by james-d-elliott** [reply]
It's a standard OAuth 2.0 Token Endpoint. It's POST (GET is invalid and expressly disallowed for OAuth 2.0 [per the spec](https://www.rfc-editor.org/rfc/rfc6749#section-3.2)) with the post data encoded as a application/x-www-form-urlencoded form. The grant should be sent via bearer scheme.

For the user data retrieval try the userinfo endpoint: https://openid.net/specs/openid-connect-core-1_0.html#UserInfo

**Reply by james-d-elliott** [reply]
It looks to me like this is super early days of this being developed on their end. The post was dated 15 days ago, and looks like it was new in that version. It also seems there may be some parts of the spec they're not super familiar with yet.

For example the above comment and the claim There is no standard for either login form URL or scopes. Consult the documentation of the solution you are using to find out valid values. but there is (maybe I'm misunderstanding what they meant):

https://www.rfc-editor.org/rfc/rfc6749#section-3.1

**Reply by tetricky** [reply]
I appreciate you following this up,  and engaging with the developer. I'm out of my depth, but it seems this will assist in a solution. Thanks!

**Reply by james-d-elliott** [reply]
I may have more time on the weekend to set it up and create an example.

**Comment by tetricky** [comment]
I'm still not sure if this is possible with the current state of misago, or how to achieve it if it is. If there are any pointers, I would appreciate them....even if it's that it can't currently be configured to work.

**Comment by tetricky** [comment]
With significant assistance from the misago author and James, we now have a working configuration for a misago integration with authelia.

I will be documenting the process and posting in misago integrations, and would also be happy to contribute documentation here.

I haven't been able to work out how to obviously make a new submission for a new integration, so would appreciate some guidance if that is desirable.

**Reply by james-d-elliott** [reply]
Thanks! I've been super busy with the next release, sorry I didn't get to this sooner. Here are all the OIDC docs: https://github.com/authelia/authelia/tree/master/docs/content/en/integration/openid-connect

Jump on chat if you're having an issue, otherwise I'm happy if you copy an existing format for say portainer (with `https://auth.example.com`  and `https://misago.example.com` or `https://forum.example.com` as the Root URLs).

**Comment by tetricky** [comment]
Nothing to apologise for. Your assistance was critical in moving the integration forward, and working together we have all got there. Me poking with a stick, you and rafalp doing all the hard work.

It is very much appreciated.

I've based my docs on the gitea integration (which I also have working), and am just working on a couple of supporting images with the correct url's.

I shall attempt to submit them to the link, adopting the format of existing files there. Please let me know if I need to do anything else.

**Comment by tetricky** [comment]
I've had a go at submitting that, but no doubt messed it up in all sorts of unimaginable ways.

https://github.com/authelia/authelia/pull/5024
